src/ConnectionPool.jl

```diff
@@ -339,7 +398,6 @@ function Base.close(c::Connection)
     if bytesavailable(c) > 0
         purge(c)
     end
-    notify(poolcondition[Threads.threadid()])
     return
 end
 
```

Comment:
The below function (purge) seems like it should be holding the writecount-lock to be valid (otherwise access to `c.io` is a potential data-race)